src/sql/engine/px/ob_dfo_scheduler.cpp

```diff
@@ -463,6 +463,14 @@ int ObSerialDfoScheduler::dispatch_sqcs(ObExecContext &exec_ctx,
                               coord_info_.interrupt_id_,
                               &sqc);
         args.set_serialize_param(exec_ctx, const_cast<ObOpSpec &>(*dfo.get_root_op_spec()), *phy_plan);
+        // 设置sqc的record_data_字段
+        ObQcRecordData& record_data = sqc.get_record_data();
+        record_data.tid = getpid();
+        record_data.sql = exec_ctx.get_sql_ctx()->cur_sql_;
+        if (record_data.sql.length() > ObQcRecordData::LIMIY_LENGTH) {
+          record_data.sql.set_length(ObQcRecordData::LIMIY_LENGTH);
+        }
```

Comment:
It's much better to use a function compared with the code directly. And probably there is another suitable position where it can avoid redundant code.

Reply:
> It's much better to use a function compared with the code directly. And probably there is another suitable position where it can avoid redundant code.

done